src/Microsoft.TestPlatform.Common/ExtensionFramework/VSExtensionManager.cs

```diff
@@ -0,0 +1,203 @@
+        private const string ExtensionManagerAssemblyName = @"Microsoft.VisualStudio.ExtensionManager";
+        private const string ExtensionManagerImplAssemblyName = @"Microsoft.VisualStudio.ExtensionManager.Implementation";
+        private const string DevenvExe = "devenv.exe";
+
+        private const string SettingsManagerTypeName = "Microsoft.VisualStudio.Settings.ExternalSettingsManager";
+        private const string SettingsManagerAssemblyName = @"Microsoft.VisualStudio.Settings.15.0";
+        
+        private IFileHelper fileHelper;
+        private string vsInstallPath;
+        private string pathToDevenv;
+
+        private Assembly extensionManagerAssembly;
+        private Assembly extensionManagerImplAssembly;
+        private Type extensionManagerServiceType;
+
+        private Assembly settingsManagerAssembly;
+        private Type settingsManagerType;
+
+        public VSExtensionManager() : this(new FileHelper())
+        {
+        }
+
+        internal VSExtensionManager(IFileHelper fileHelper)
+        {
+            this.fileHelper = fileHelper;
+        }
+
+        /// <summary>
+        /// Get the available unit test extensions installed.
+        /// If no extensions are installed then it returns an empty list.
+        /// </summary>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate"), System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public IEnumerable<string> GetUnitTestExtensions()
+        {
+            try
+            {
+                return GetTestExtensionsInternal(Constants.UnitTestExtensionType);
+            }
+            catch (Exception ex)
+            {
+                string message = string.Format(CultureInfo.CurrentCulture, Resources.FailedToFindInstalledUnitTestExtensions, ex.Message);
+                throw new TestPlatformException(message, ex);
+            }
+        }
+
+        /// <summary>
+        /// Get the unit tests extensions
+        /// </summary>
+        private IEnumerable<string> GetTestExtensionsInternal(string extensionType)
+        {
+            IEnumerable<string> installedExtensions = new List<string>();
+
+            // Navigate to the IDE folder
+            this.vsInstallPath = new DirectoryInfo(typeof(ITestPlatform).GetTypeInfo().Assembly.GetAssemblyLocation()).Parent.Parent.Parent.FullName;
+            this.pathToDevenv = Path.Combine(vsInstallPath, DevenvExe);
+
+            // Check for devenv
+            if (!this.fileHelper.Exists(this.pathToDevenv))
+            {
+                throw new TestPlatformException(string.Format(CultureInfo.CurrentCulture, Resources.VSInstallationNotFound, this.vsInstallPath));
+            }
+
+            // Adding resolution paths for resolving dependencies.
+            var resolutionPaths = new List<string>() {
```

Comment:
If the set is constant, why not use an array?

Reply:
good point.